### DISCUSSION TITLE
Spec Issue: Do not forward

### DISCUSSION THREAD

**Discussion by JackAAnders** [christianTF/LoxBerry-Plugin-MQTT-Gateway#66] [Q&A]
Hi Christian,

I had the issue that outputs were changing all the time. Which led to the blinds moving up and down all the time.
When using an analog output in Loxone to publish a topic, it lead to changing values, although the output was driven with a constant value in Loxone. Finally I found that the "do not forward" option prevents the current value of beeing feed into the output again. 
I would suggest to recommend that the do not forward option is activated for analog outputs from loxone.

After I had activated the option this feedback loop was broken up and everything was working as expected.

**Comment by christianTF** [comment]
Hello Jack,
is the name of the _output_ equal to the name the Gateway is returning the data to? Then, you have created a loop.
_Do not name a **publishing output** equal to the name the Gateway is sending to._ Use any other name for the output. 

In general, it is hard to give recommendations as I don't know every data passing the Gateway, so I cannot say if the recommendation is valid for every use-case.

**Reply by JackAAnders** [reply]
Thanks Christian,
this clarified where the issue was coming from. Indeed, I did name the publishing output corresponding to its topic. For me it felt naturally to do this way. MQTT only knows topics and weather you input data or output data is defined by the way you access it.

Maybe I was expecting that the Loxone output does only publish, but it also cares about its return value. A warning about the naming of outputs would be appreciated. I like to give the same things the same name and hence do not want another naming scheme for inputs and outputs.

Other than that, it works great! Thanks a ton.

Regards,
Jack

**Comment by christianTF** [comment]
In fact, the MQTT Gateway can address every Loxone block by it's name. If you have a switch block that receives 1 from the Gateway, it will turn on. Or you have a Counter block with the "VI name" from the Gateway, it will accept the MQTT analogue value.
Regards, Christian
